### DISCUSSION TITLE
Floating window hides text

### DISCUSSION THREAD

**Discussion by N19htfox** [Isrothy/neominimap.nvim#193] [Q&A] [answered]
Hey, I love this plugin but I have one issue.
When I have a line that gets past a certain length, the floating window for the minimap hides the text and with the cursor on the end of the line, I can't scroll over far enough to bring the text past the minimap and into view.
![image](https://github.com/user-attachments/assets/80e3c04b-20db-41a7-89d1-d5729a4bb41b)

Is there a way to make this work in the vanilla plugin? Or could the functionality be altered slightly to make the scrolling behave more nicely?

The functionality I would like to achieve would be when the cursor goes over the minimap, the plugin could automatically scroll to move the cursor back out from behind the floating window.
![image](https://github.com/user-attachments/assets/6b4ec763-e49d-447b-b44f-b20b2bac2e08)

**Comment by Isrothy** [comment, marked as answer]
Put this in your Neovim configuration.
```lua
vim.opt.sidescrolloff = 36 -- A value that is larger then the width of minimap
```

**Reply by N19htfox** [reply]
I fixed it by adding this to my settings.lua (required after my minimap setup)
``` lua
-- Horiontal scroll margin ( Take minimap into account ) --
if vim.g.neominimap.layout == 'float' then
    vim.opt.sidescrolloff = vim.g.neominimap.float.minimap_width + 2
else
    vim.opt.sidescrolloff = 5
end
```

Worked like a charm thanks!
